Make sure not to have problems with trying to use libiconv without including
its header.

Preprocessor conditionals taken from SDL 1.2.14.

--- $sourcebase.orig/include/SDL_stdinc.h	2007-12-31 05:48:36.000000000 +0100
+++ $sourcebase/include/SDL_stdinc.h	2010-02-19 17:33:54.000000000 +0100
@@ -68,7 +68,7 @@
 #ifdef HAVE_CTYPE_H
 # include <ctype.h>
 #endif
-#ifdef HAVE_ICONV_H
+#if defined(HAVE_ICONV) && defined(HAVE_ICONV_H)
 # include <iconv.h>
 #endif
 
@@ -569,7 +569,7 @@
 #define SDL_ICONV_EILSEQ	(size_t)-3
 #define SDL_ICONV_EINVAL	(size_t)-4
 
-#ifdef HAVE_ICONV
+#if defined(HAVE_ICONV) && defined(HAVE_ICONV_H)
 #define SDL_iconv_t     iconv_t
 #define SDL_iconv_open  iconv_open
 #define SDL_iconv_close iconv_close
